Src/Newtonsoft.Json/Converters/XmlNodeConverter.cs

```diff
@@ -1177,6 +1177,11 @@ private bool IsArray(IXmlNode node)
 
         private void SerializeGroupedNodes(JsonWriter writer, IXmlNode node, XmlNamespaceManager manager, bool writePropertyName)
         {
+            if (node.ChildNodes.Count == 0)
```

Comment:
Good idea. I'm not sure it's worth doing for attributes though, as it being not-null but empty requires hitting the same collection anyway so it's a bit swing and roundabouts, and it being null is already tested for.